python_modules/dagster/dagster/_cli/utils.py

```diff
@@ -1,40 +1,109 @@
+import logging
 import os
 import tempfile
 from contextlib import contextmanager
-from typing import Callable, Iterator, Optional
-
-import click
+from typing import Iterator, Optional
 
 from dagster._core.instance import DagsterInstance, InstanceRef
 from dagster._core.instance.config import is_dagster_home_set
+from dagster._core.secrets.env_file import get_env_var_dict
+
+
+def _load_env_var_dict(logger: logging.Logger) -> None:
+    dotenv_file_path = os.getcwd()
+
+    env_var_dict = get_env_var_dict(dotenv_file_path)
+
+    # Load any env vars
+    for k, v in env_var_dict.items():
+        os.environ[k] = v
+
+    if len(env_var_dict):
+        logger.info(
+            "Loaded environment variables from .env file: "
+            + ",".join([env_var for env_var in env_var_dict]),
+        )
```

Comment:
IMO this should be a context manager that unsets the environment variables on exit for tests etc